color-picker-project/src/components/colorpicker/ColorPicker.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+import { hsvaToRgbaString } from '@uiw/color-convert';
+import Wheel from '@uiw/react-color-wheel';
+import ShadeSlider from '@uiw/react-color-shade-slider';
+import Alpha from '@uiw/react-color-alpha';
+import Material from '@uiw/react-color-material';
+import EditableInputHSLA from '@uiw/react-color-editable-input-hsla';
+import './ColorPicker.css'
+
+function ColorPicker ({hsva, setHsva, hex, setHex}) {
+
+    return (
+      <div className="color-picker-div">
+          <div className="color-picker-container">
+            <h2 className="color-picker-title">Choose a color...</h2>
+            <Wheel className="color-picker-wheel" color={hsva} onChange={(color) => setHsva({ ...hsva, ...color.hsva })} />
+            <ShadeSlider
+            hsva={hsva}
+            style={{ width: 210, marginTop: 20 }}
+            onChange={(newShade) => {
+              setHsva({ ...hsva, ...newShade });
+            }}
+            />
+            <Material className="color-picker-material" style={{ width: '15%' }}
+              color={hsva}
+              onChange={(color) => {
+                setHsva(color.hsva), setHex(color.hex);
+              }}
+            />
+            <Alpha style= {{ width: '25%' }} className="color-picker-alpha"
+              hsva={hsva}
+              onChange={(newAlpha) => {
+                setHsva({ ...hsva, ...newAlpha });
+              }}
+            />
+            <EditableInputHSLA className="color-picker-hsla"
```

Comment:
This is not a high priority, but I'm curious why the main state variable is hsva when there isn't a UI input for the hsva format.  I poked around the [docs](https://uiwjs.github.io/react-color/#/editable-input-hsla) a little bit and I'm not sure that React Color even supports an hsva UI element (unless I'm missing something).

It just seems weird that every color format in the UI has to be converted back and forth, but I can't tell if that's just a quirk of React Color or has to do with how the component was set up.